chain/api/src/config.rs

```diff
@@ -165,10 +151,11 @@ pub struct ChainNetworkConfig {
 /// Check whether the version is allowed
 fn satisfies(version: &str, allowed_versions: &str) -> crate::errors::Result<bool> {
     let allowed_versions = VersionReq::parse(allowed_versions)
-        .map_err(|e| HoprLibError::GeneralError(format!("failed to deserialize allowed version string: {}", e)))?;
+        .map_err(|e| HoprChainError::Configuration(format!("failed to deserialize allowed version string: {}", e)))?;
 
-    let version = Version::from_str(version)
-        .map_err(|e| HoprLibError::GeneralError(format!("failed to deserialize current lib version string: {}", e)))?;
+    let version = Version::from_str(version).map_err(|e| {
+        HoprChainError::Configuration(format!("failed to deserialize current lib version string: {}", e))
+    })?;
 
     Ok(allowed_versions.matches(&version))
```

Comment:
Careful, that this will not match version with additional information like `-rc.3.commit-abcdef`, unless the additional information from the version is stripped of to contain just major, minor and patch.

Reply:
It uses the `crate::constants::APP_VERSION_COERCED` from hopr-lib, which has the `<major>.<minor>.<patch>` format.